pallets/loans/src/entities/pricing/external.rs

```diff
@@ -318,3 +320,110 @@ impl<T: Config> ExternalActivePricing<T> {
 		Ok(())
 	}
 }
+
+/// Adds `with_linear_pricing` to ExternalPricing struct for migration to v4
+pub mod v3 {
+	use cfg_traits::{interest::InterestRate, Seconds};
+	use parity_scale_codec::{Decode, Encode};
+
+	use crate::{
+		entities::{
+			interest::ActiveInterestRate,
+			pricing::{external::MaxBorrowAmount, internal, internal::InternalActivePricing},
+		},
+		pallet::AssetOf,
+		types::{LoanRestrictions, RepaymentSchedule},
+		Config,
+	};
+
+	#[derive(Encode, Decode)]
+	pub struct CreatedLoan<T: Config> {
+		info: LoanInfo<T>,
+		borrower: T::AccountId,
+	}
+
+	#[derive(Encode, Decode)]
+	pub struct LoanInfo<T: Config> {
+		pub schedule: RepaymentSchedule,
+		pub collateral: AssetOf<T>,
+		pub interest_rate: InterestRate<T::Rate>,
+		pub pricing: Pricing<T>,
+		pub restrictions: LoanRestrictions,
+	}
+
+	#[derive(Encode, Decode)]
+	pub enum Pricing<T: Config> {
+		Internal(internal::InternalPricing<T>),
+		External(ExternalPricing<T>),
+	}
+
+	#[derive(Encode, Decode)]
+	pub struct ExternalPricing<T: Config> {
+		pub price_id: T::PriceId,
+		pub max_borrow_amount: MaxBorrowAmount<T::Quantity>,
+		pub notional: T::Balance,
+		pub max_price_variation: T::Rate,
+	}
+
+	impl<T: Config> CreatedLoan<T> {
+		pub fn migrate(self, with_linear_pricing: bool) -> crate::entities::loans::CreatedLoan<T> {
+			let pricing = match self.info.pricing {
+				Pricing::Internal(i) => crate::entities::pricing::Pricing::Internal(i),
+				Pricing::External(e) => {
+					crate::entities::pricing::Pricing::External(e.migrate(with_linear_pricing))
+				}
+			};
+			let info = crate::entities::loans::LoanInfo::<T> {
+				pricing,
+				schedule: self.info.schedule,
+				collateral: self.info.collateral,
+				interest_rate: self.info.interest_rate,
+				restrictions: self.info.restrictions,
+			};
+
+			crate::entities::loans::CreatedLoan::<T>::new(info, self.borrower)
+		}
+	}
+
+	#[derive(Encode, Decode)]
+	pub enum ActivePricing<T: Config> {
+		Internal(InternalActivePricing<T>),
+		External(ExternalActivePricing<T>),
+	}
+
+	#[derive(Encode, Decode)]
+	pub struct ExternalActivePricing<T: Config> {
+		info: ExternalPricing<T>,
+		outstanding_quantity: T::Quantity,
+		pub interest: ActiveInterestRate<T>,
+		latest_settlement_price: T::Balance,
+		settlement_price_updated: Seconds,
+	}
+
+	impl<T: Config> ExternalActivePricing<T> {
+		pub fn migrate(self, with_linear_pricing: bool) -> super::ExternalActivePricing<T> {
+			super::ExternalActivePricing {
+				info: self.info.migrate(with_linear_pricing),
+				outstanding_quantity: self.outstanding_quantity,
+				interest: self.interest,
+				latest_settlement_price: self.latest_settlement_price,
+				settlement_price_updated: self.settlement_price_updated,
```

Comment:
We need to ensure here that `settlement_price_updated` does no exceed the maturity date. [ref](https://github.com/centrifuge/centrifuge-chain/pull/1842/files#diff-637356fbd7809c17265d6b8ab519abc2b0ffa4df672cfb33a321795190f441baR437)

Reply:
Never mind, after the last fix update. You do not need to do anything in this PR regarding this.